catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/SearchFormsLoader.java

```diff
@@ -0,0 +1,223 @@
+
+/**
+ * Loads initial system template configuration from the file system so template defaults in
+ * distributions can vary independently.
+ */
+public class SearchFormsLoader implements Supplier<List<Metacard>> {
+  private static final Logger LOGGER = LoggerFactory.getLogger(SearchFormsLoader.class);
+
+  private static final File DEFAULT_FORMS_DIRECTORY =
+      new File(new AbsolutePathResolver("etc/forms").getPath());
+
+  private static final String FORMS_FILE_NAME = "forms.json";
+
+  private static final String RESULTS_FILE_NAME = "results.json";
+
+  private final File configDirectory;
+
+  public static Supplier<List<Metacard>> config() {
+    return new SearchFormsLoader();
+  }
+
+  public SearchFormsLoader() {
+    this(DEFAULT_FORMS_DIRECTORY);
+  }
+
+  public SearchFormsLoader(File configDirectory) {
+    this.configDirectory = configDirectory;
+  }
+
+  @Override
+  public List<Metacard> get() {
+    if (!configDirectory.exists()) {
+      LOGGER.debug("Could not locate forms directory");
+      return Collections.emptyList();
+    }
+
+    // What's our policy here? Do we want the system to blowup / make noise if this isn't possible?
+    // Especially due to the security manager?
+    if (!configDirectory.canRead()) {
+      LOGGER.debug("Forms directory exists but could not be read");
+      return Collections.emptyList();
+    }
+
+    File formsFile = configDirectory.toPath().resolve(FORMS_FILE_NAME).toFile();
+    File resultsFile = configDirectory.toPath().resolve(RESULTS_FILE_NAME).toFile();
+
+    return Stream.concat(
+            loadFile(formsFile, this::formMapper), loadFile(resultsFile, this::resultsMapper))
+        .collect(Collectors.toList());
+  }
+
+  /** Read the provided JSON file and return a stream of metacards. */
+  private Stream<Metacard> loadFile(File file, Function<? super Map, Metacard> mapper) {
+    if (!file.exists()) {
+      LOGGER.debug("Could not locate {}", file.getName());
+      return Stream.empty();
+    }
+
+    String payload = getFileContent(file);
+    if (payload == null) {
+      LOGGER.debug("Problem reading {}", file.getName());
+      return Stream.empty();
+    }
+
+    try {
+      return Stream.of(Boon.fromJson(payload))
+          .map(List.class::cast)
+          .flatMap(List::stream)
+          .filter(Map.class::isInstance)
+          .map(Map.class::cast)
+          .map(mapper)
+          .filter(Objects::nonNull)
+          // Invoke a terminal operation so the exception can be handled here
+          .collect(Collectors.toList())
+          .stream();
+    } catch (ClassCastException e) {
+      LOGGER.warn(
+          "Could not load forms configuration in {}, JSON should be a list of maps",
+          file.getName());
+      return Stream.empty();
+    }
+  }
+
+  /** Parse the JSON map for initializing system form templates. */
+  @Nullable
+  private Metacard formMapper(Map map) {
+    LOGGER.debug("Starting form processing...");
```

Comment:
This does strike me as a very "trace-y" message. I might rework the message so it can work at the `debug` level. We rarely ever use trace. It's utility is pretty scarce. 

I think if the metacard ID is included then this makes sense as `debug` so troubleshooters have all the context they need at the same log level.

Reply:
Even with the ID, this is still a `trace` message as it is tracking program execution.  https://codice.atlassian.net/wiki/spaces/DDF/pages/7045131/Log+Levels.  I agree with @jrnorth

Reply:
I'd probably remove it completely instead of pushing it down to trace. Thoughts on that?

Reply:
👍